src/background.js

```diff
@@ -266,33 +266,69 @@ async function maybeReopenTab (url, tab, request) {
   return {cancel: true};
 }
 
+function getRootDomain(hostname) {
+  let parts = hostname.split(".");
+
+  // Default array length of hostname split
+  let defaultDomainLength = 3;
+
+  // If there's no subdomain, return hostname as is.
+  if (parts.length < defaultDomainLength) {
+    return hostname;
+  }
+
+  let lastArrayLength = parts[(parts.length - 1)].length;
+
+
+  if (lastArrayLength < 3) {
+    // Returns true if last part of hostname is two-characters, meaning it's a
+    // country-level domain.
+    let nextToLastItem = parts[(parts.length - 2)];
+    if (nextToLastItem === "co" || nextToLastItem === "com") {
+      // Checks to see if a third-level domain is present.
+      // Example: https://www.amazon.co.uk returns "amazon.co.uk"
+      defaultDomainLength = 4;
+    }
+  }
+
+  while (parts.length > (defaultDomainLength - 1)) {
+    // Trim around down to final domain
+    parts.shift();
+  }
+
+  let rootDomain = parts.join(".");
+  return rootDomain;
+}
```

Comment:
Might see about using [the `psl` library](https://www.npmjs.com/package/psl) for this? Could grab the `psl.min.js` file and include it before `background.js`? It looks like `psl` is even already part of our dependencies? (Though I can't tell if we're using it anywhere else?